lib/trax/model/struct.rb

```diff
@@ -122,6 +103,82 @@ class << self
       def value
         self
       end
+
+      private
+      #this only supports properties 1 level deep, but works beautifully
+      #I.E. for this structure
+      # define_attributes do
+      #   struct :custom_fields do
+      #     enum :color, :default => :blue do
+      #       define :blue,     1
+      #       define :red,      2
+      #       define :green,    3
+      #     end
+      #   end
+      # end
+      # ::Product.by_custom_fields_color(:blue, :red)
+      # will return #{Product color=blue}, #{Product color=red}
+      def self.define_scopes_for_enum(attribute_name, enum_klass)
+        return unless has_active_record_ancestry?(enum_klass)
+
+        model_class = model_class_for_property(enum_klass)
+        field_name = enum_klass.parent_definition.name.demodulize.underscore
+        attribute_name = enum_klass.name.demodulize.underscore
+        scope_name = :"by_#{field_name}_#{attribute_name}"
+        model_class.scope(scope_name, lambda{ |*_scope_values|
+          _integer_values = enum_klass.select_values(*_scope_values.flat_compact_uniq!)
+          model_class.where("#{field_name} -> '#{attribute_name}' IN(#{_integer_values.to_single_quoted_list})")
```

Comment:
Tell me about the safety of `#to_single_quoted_list`. Concatenating into SQL concerns me. Doesn't active record allow placeholders with safe escaping?